pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -200,8 +200,21 @@ protected void validateSuperUserAccess() {
                 }
                 log.debug("Successfully authorized {} (proxied by {}) as super-user",
                           originalPrincipal, appId);
-            } else if (!config().getSuperUserRoles().contains(appId)) {
-                throw new RestException(Status.UNAUTHORIZED, "This operation requires super-user access");
+            } else {
+                CompletableFuture<Boolean> isSuperUserFuture;
+                try {
+                    isSuperUserFuture = pulsar.getBrokerService()
+                            .getAuthorizationService()
+                            .isSuperUser(appId);
+
+                    if (!isSuperUserFuture.get()) {
+                        throw new RestException(Status.UNAUTHORIZED, "This operation requires super-user access");
+                    }
+                } catch (InterruptedException | ExecutionException e) {
```

Comment:
Using `.join()` instead of `.get()` would simplify the code here. In any case an exception thrown will trigger a 500 error.
eg: 

```java
if (!pulsar.getBrokerService()
    .getAuthorizationService()
    .isSuperUser(appId)
    .join()) {
    throw new RestException(Status.UNAUTHORIZED, "This operation requires super-user access");
}

```

Reply:
thansk for the suggestion, this looks much better